src/OddJobs/Job.hs

```diff
@@ -424,6 +435,26 @@ runJob jid = do
       liftIO $ void $ Prelude.foldr tryHandler (throwIO e) handlers
       pure ()
 
+killJob :: (HasJobRunner m) => JobId -> m ()
+killJob jid = do
+  threadsRef <- envJobThreadsRef <$> getRunnerEnv
+  threads <- liftIO $ readIORef threadsRef
+  mJob <- findJobById jid
+
+  case (mJob, jid `DM.lookup` threads) of
+    (Just job, Just thread) -> do
+      log LevelInfo $ LogKillJobSuccess job
+      void $ finally
+        (uninterruptibleCancel thread)
+        (atomicModifyIORef' threadsRef $ \threads -> (DM.delete jid threads, ()))
+
+    (Just job, Nothing) -> do
+      log LevelInfo $ LogKillJobFailed job
+      return ()
+
+    (Nothing, _) ->
+      error $ "Unable to find job in db to kill, jobId = " <> (show jid)
```

Comment:
I'm assuming you have used `error` here because the case should be impossible - and looking at the control flow, I think I agree. But I think it would be better to `log LevelError` here, just so the job runner doesn't crash if something unexpected happens or some other part of the polling logic changes.

Reply:
I Agree, will make the change.